internal/index_read.go

```diff
@@ -165,15 +166,22 @@ func readIntervals(r io.Reader, typ string) ([]bgzf.Offset, error) {
 	if n == 0 {
 		return nil, nil
 	}
-	var vOff uint64
 	offsets := make([]bgzf.Offset, n)
-	for i := range offsets {
-		err := binary.Read(r, binary.LittleEndian, &vOff)
+	// chunkSize determines the number of offsets consumed by each binary.Read.
+	const chunkSize = 512
+	var vOffs [chunkSize]uint64
+	for i := 0; i < int(n); i += chunkSize {
+		// Limit the size of vOffs on final iteration.
+		l := min(int(n)-i, chunkSize)
+		err := binary.Read(r, binary.LittleEndian, vOffs[:l])
```

Comment:
`l` doesn't add anything to readability here and is not reused, so do the calculation inline.

Also `err` is already declared, so we don't need `:=`.

The comment does not add anything since the `min(n, len(arr))` which is functionally what the expression is idiomatic for this behaviour (maybe the expression could be changed to `min(int(n)-1, len(vOffs))` to reflect this - note that the assembly that is generated should be identical since `len([nt]T)` is a compile time constant).